src/shared/settingsholder.cpp

```diff
@@ -296,6 +300,8 @@ bool SettingsHolder::rollbackTransaction() {
     QMetaObject::invokeMethod(this, i.value().first, Qt::DirectConnection);
   }
 
+  emit transactionRolledBack();
```

Comment:
Instead of this, can you do this: `qScopeGuard([&] { emit transactionRolledBack(); }` before `if (!finalizeTansaction()` ?

In this way, if something goes wrong, we still emit the transactionRolledBack.